container/build_tar.py

```diff
@@ -191,18 +228,29 @@ def add_deb(self, deb):
     Raises:
       DebError: if the format of the deb archive is incorrect.
     """
+    pkg_data_found = False
+    pkg_metadata_found = False
     with archive.SimpleArFile(deb) as arfile:
       current = arfile.next()
-      while current and not current.filename.startswith('data.'):
+      while current:
+        if current.filename.startswith('data'):
+          pkg_data_found = True
+          # Add pkg_data to image tar
+          with self.write_temp_file(suffix=os.path.splitext(current.filename)[-1], data=current.data) as tmpfile:
+            self.add_tar(tmpfile)
+        elif current.filename.startswith('control'):
+          pkg_metadata_found = True
+          # Add metadata file to image tar
+          with self.write_temp_file(suffix=os.path.splitext(current.filename)[-1], data=current.data) as tmpfile:
+            self.add_pkg_metadata(metadata_tar=tmpfile, deb=deb)
         current = arfile.next()
-      if not current:
-        raise self.DebError(deb + ' does not contains a data file!')
-      tmpfile = tempfile.mkstemp(suffix=os.path.splitext(current.filename)[-1])
-      with open(tmpfile[1], 'wb') as f:
-        f.write(current.data)
-      self.add_tar(tmpfile[1])
-      os.remove(tmpfile[1])
 
+    if not pkg_data_found:
+      raise self.DebError(deb + ' does not contains a data file!')
+    if not pkg_metadata_found:
+      #TODO(tejaldesai)should we raise an exception or create metadata file with package name as
+      # os.path.basename(deb)
+      raise self.DebError(deb + ' does not contains a data file!')
```

Comment:
Can you give two examples of what these might be?  Are they always going to be different?  Might be tough for downstream systems to check non canonical names.

Reply:
Examples are these would be. 
- goobuntu-puppet-external_3762~trusty_all
- curl_7.35.0-1ubuntu2.12_amd64
They do follow some convention:
<pkg_nm>_<version><os>_<platform>
 But if you end up building a .deb locally and want to test it in your docker image, this name can be anything.
My proposal is to not handle name, version parsing from the string and just write it as is.
So for a test_package_1-2-0.deb will look something like this:
```
tar -xf temp_control.tar control 
cat control
Package: test_package_1-2-0
```
This will create a dpkg status file as 
/var/lib/dpkg/status.d/test_package_1-2-0

How will this help Drydock? 
We will have a record packages installed.

Reply:
Or we can fail like what i have now and make sure the developer creates a dpkg package with valid metadata bundled.